packages/@webex/plugin-cc/test/unit/spec/cc.ts

```diff
@@ -10,21 +12,79 @@ describe('CC Tests', () => {
     webex = new MockWebex({
       children: {
         cc: ContactCenter,
-        mercury: Mercury,
         services: Services,
       },
     });
-  })
 
-  it('Success: invoke the register function, resolved promise', () => {
-    webex.cc.register(true).then((data) => {
-        expect(data).toEqual('Success: Dummy data returned');
-    })
-  });
+    webex.cc.webSocket = new WebSocket(
+      {
+        parent: webex,
+      }
+    );
+
+    webex.cc.webSocket.on = jest.fn((event, callback) => {
+      if (event === 'event') {
+        // Store the callback to call it later
+        webex.cc.webSocket._eventCallback = callback;
+      }
+    });
+
+    // Mock the establishConnection function to trigger the event with Welcome type
+    webex.cc.webSocket.subscribeAndConnect = jest.fn().mockImplementation(() => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const event = {
+            type: CC_EVENTS.WELCOME,
+            data: {
+              agentId: 'dummy-agent-id'
+            }
+          };
+          if (webex.cc.webSocket._eventCallback) {
+            webex.cc.webSocket._eventCallback(event); // Trigger the stored event callback
+          }
+          resolve('Success: Dummy data returned');
+        }, 100); // Simulate the event being fired after 100ms
+      });
+    });
+
+    // Mock the register function to call establishConnection and handle Welcome event
+    webex.cc.register = jest.fn().mockImplementation(() => {
```

Comment:
I would like to discuss testing approach

Reply:
addressed